hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieUnmergedFileGroupRecordBuffer.java

```diff
@@ -77,7 +76,7 @@ protected boolean doHasNext() throws IOException {
       if (nextRecordInfo.getLeft().isPresent()) {
         nextRecord = nextRecordInfo.getKey().get();
       } else {
-        nextRecord = readerContext.constructRawDeleteRecord(nextRecordInfo.getRight());
+        throw new IllegalStateException("No deletes should exist in unmerged reading mode");
```

Comment:
Can deletes from data log blocks still enter here?

Reply:
Let's also create a Jira to track the delete block support in the file group record buffer.

Reply:
https://issues.apache.org/jira/browse/HUDI-8414